beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -115,8 +115,27 @@ func (f *ForkChoice) InsertNode(ctx context.Context, state state.BeaconState, ro
 		return errNilBlockHeader
 	}
 	parentRoot := bytesutil.ToBytes32(bh.ParentRoot)
-	var payloadHash [32]byte
-	if state.Version() >= version.Bellatrix {
+	var payloadHash, parentHash [32]byte
+	if state.Version() >= version.EPBS {
+		slot, err := state.LatestFullSlot()
+		if err != nil {
+			return err
+		}
+		latestHash, err := state.LatestBlockHash()
+		if err != nil {
+			return err
+		}
+		if slot == state.Slot() {
+			latestHeader, err := state.LatestExecutionPayloadHeaderEPBS()
+			if err != nil {
+				return err
+			}
+			copy(payloadHash[:], latestHash)
+			copy(parentHash[:], latestHeader.ParentBlockHash)
+		} else {
+			copy(parentHash[:], latestHash)
+		}
```

Comment:
Not sure if you agree with me here but I thought this is more clear. Also in the same slot case, you don't need to call `state.LatestBlockHash()`

```suggestion
		slot, err := state.LatestFullSlot()
		if err != nil {
			return err
		}
		if slot == state.Slot() {
			latestHeader, err := state.LatestExecutionPayloadHeaderEPBS()
			if err != nil {
				return err
			}
			copy(payloadHash[:], latestHeader.BlockHash)
			copy(parentHash[:], latestHeader.ParentBlockHash)
		} else {
			latestHash, err := state.LatestBlockHash()
			if err != nil {
				return err
			}
			copy(parentHash[:], latestHash)
		}
```